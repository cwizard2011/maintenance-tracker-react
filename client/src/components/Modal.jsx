import React, { Component } from 'react';
import PropTypes from 'prop-types';

/* eslint class-methods-use-this: [0, { "exceptMethods": ["buggerToggle"] }] */
/* eslint no-unused-expressions: [0, { "allowTernary": true }] */
// eslint-disable-next-line react/prefer-stateless-function
class Modal extends Component {
  onClose(e) {
    this.props.onClose && this.props.onClose(e);
  }
  render() {
    if (!this.props.show && !this.props.login) {
      return null;
    }
    return (
      <div>
        <div className="modals">
          <div className="modals-content form-field modals-body">
            <span role="presentation" className="close" onClick={(e) => { this.onClose(e); }}>&times;</span>
            {this.props.children}
          </div>
        </div>
      </div>
    );
  }
}
Modal.propTypes = {
  children: PropTypes.node.isRequired,
  show: PropTypes.bool.isRequired,
  login: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
};
export default Modal;
