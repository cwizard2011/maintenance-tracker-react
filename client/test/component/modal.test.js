import React from 'react';
import { shallow } from 'enzyme';
import Modal from '../../src/components/Modal';

const props = {
  onClose: jest.fn()
};
describe('Modal', () => {
  it('not render Modal', () => {
    const wrapper = shallow(<Modal />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should display modal on click', () => {
    const wrapper = shallow(<Modal {...props} />);
    const instance = wrapper.instance();
    instance.onClose();
    expect(props.onClose).toHaveBeenCalled();
  });
  it('should render modal correctly', () => {
    const newProps = {
      ...props,
      show: true,
      children: 'new'
    };
    const wrapper = shallow(<Modal {...newProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').length).toEqual(3);
  });
});
