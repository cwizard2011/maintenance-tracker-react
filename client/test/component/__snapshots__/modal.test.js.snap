// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal not render Modal 1`] = `""`;

exports[`Modal should render modal correctly 1`] = `
<div>
  <div
    className="modals"
  >
    <div
      className="modals-content form-field modals-body"
    >
      <span
        className="close"
        onClick={[Function]}
        role="presentation"
      >
        Ã—
      </span>
      new
    </div>
  </div>
</div>
`;
